### DISCUSSION TITLE
How are releases triggered?

### DISCUSSION THREAD

**Discussion by kevkearney** [aws/aws-cdk#17006] [Q&A] [answered]
Just a general curiousity. How are releases triggered? Is there some admin of the project that manually cuts a release when they feel it's necessary, or is there a set of automated rules. I looked in the contributing docs but couldn't find the answer.

**Comment by skinny85** [comment, marked as answer]
The releases are fully automated after being triggered by a human. One of the engineers on the CDK team triggers that process once a week.

In theory, anyone can start a release, as it's kicked off by a Pull Request to the `release` branch, [like this](https://github.com/aws/aws-cdk/pull/16969). But in practice, we would probably not approve a PR like that opened by a contributor, as we want our automation to create that PR (as that makes sure all of the necessary steps for the release have been executed correctly - it basically boils down to running the [`bump.sh` script](https://github.com/aws/aws-cdk/blob/fbb49fec0e70dbec119bdb98562d68ccc2f0385d/bump.sh#L1-L23) on a new branch, but the less a human has to do that, the smaller the chance of them screwing up, and this includes our team members too 🙂).

Hope that clears this up!

Thanks,
Adam
